.github/workflows/test_github_app.yml

```diff
@@ -0,0 +1,44 @@
+name: test-github_app
+
+on:
+  push:
+  #   branches:
+  #     - "!main"
+  # pull_request:
+  workflow_call:
+  workflow_dispatch:
+
+jobs:
+  test-implementation-job:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      # To use this repository's private action, you must check out the repository
+      - name: token-generation
+        uses: actions/create-github-app-token@v1
+        id: source-app-token
+        with:
+          app-id: ${{ secrets.TEST_GITHUB_APP_ID }}
+          private-key: ${{ secrets.TEST_APP_PEM_FILE }}
+          owner: alexvanderberkel
+          repositories: private-test-repo
+      
+      - name: Checkout
+        # https://github.com/actions/checkout#usage
+        uses: actions/checkout@v4
+        with:          
+          token: ${{ steps.source-app-token.outputs.token }}             
+      
+      - name: Test action step
+        uses: ./ # Uses an action in the root directory
+        env:
+          MY_VAR: "bar"
+        with:
+          source_repo_path: alexvanderberkel/private-test-repo
+          source_gh_token: ${{ steps.source-app-token.outputs.token }}     
+          target_gh_token: ${{ github.token }}      
```

Comment:
actually you should not need the `target_gh_tocken`. Can you remove that for a test?